script/universal/NestedMultisigBuilder.sol

```diff
@@ -40,6 +40,34 @@ abstract contract NestedMultisigBuilder is MultisigBase {
      * -----------------------------------------------------------
      */
 
+    // Virtual method which can be overwritten.
+    // This allows different nonce overrides for each safe in the nested multisig case.
+    // IMPORTANT: this method is used in the sign, simulate, AND execution contexts
+    // If you override it, ensure that the behavior is correct for all contexts
+    // As an example, if you are pre-signing a message that needs safe.nonce+1 (before safe.nonce is executed),
+    // you should explicitly set the nonce value with an env var.
+    // Overwriting this method with safe.nonce + 1 will cause issues upon execution because the transaction
+    // hash will differ from the one signed.
+    function _getNonce(IGnosisSafe safe) internal view override virtual returns (uint256 nonce) {
+        string memory safeAddrStr = vm.toString(address(safe));
+        nonce = safe.nonce();
+        console.log("Safe", safeAddrStr, "current nonce:", nonce);
+
+        // In this overridden method, the process for determining the nonce is as follows:
+        //   1. We look for an env var of the name SAFE_NONCE_{UPPERCASE_SAFE_ADDRESS}. For example,
+        //      SAFE_NONCE_0X6DF4742A3C28790E63FE933F7D108FE9FCE51EA4.
+        //   2. If it exists, we use it as the nonce override for the safe.
+        //   3. If it does not exist, we use the current nonce of the safe.
+        //   4. We explicitly do not use SAFE_NONCE as a fallback, because in the nested case it is
+        //      ambiguous which safe it refers to.
+        string memory safeNonceEnvVarName = string.concat("SAFE_NONCE_", vm.toUppercase(safeAddrStr));
+        try vm.envUint(safeNonceEnvVarName) {
+            nonce = vm.envUint(safeNonceEnvVarName);
+            console.log("Creating transaction with nonce:", nonce);
+        }
+        catch {}
+    }
```

Comment:
With this schema it's only possible to create one overridden nonce per safe address. This strictly means that we can only pre-sign up to two transactions. Is this sufficient for what you need this for?